### ISSUE TITLE
[Time field] Validation message not announced by screen reader

### ISSUE THREAD

**Issue by DanielleRameau** [sl-design-system/components#2740]
### Provide a general summary of the issue here

**Description:**
When the time field is marked as required, the error message is not announced by screen readers (e.g., NVDA). This creates an accessibility gap for users relying on assistive technologies.

**Steps to Reproduce:**
- Leave the required time field empty and submit the form.
- Enter an invalid time (e.g., 07:99) and submit.
- Use NVDA to navigate the form.
- Observe that the error message is not read aloud.

Screenshot:

<img width="1465" height="603" alt="Image" src="https://github.com/user-attachments/assets/4b0f408b-b0a7-4a3d-b2bb-81f6f5e86859" />

**Suggested Fix:**
- probably we need to add a separate message when the field is empty and different one when the wrong value was typed in e.g. '07:99';
- we also need to check why the validation message is not read by NVDA and also make sure it works in different form controls.

Issue found during internal a11y audit: https://github.com/sl-design-system/components/issues/2580#issuecomment-3376035899

### 👤 Your name

DR

### 🧢 Your Product/Team

SLDS
